modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +361,250 @@ public void testGetAllIdPs() throws Exception {
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All filtered paginated idp list size : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            IdentityProvider notFilteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (idp.getIdentityProviderName().equals(testIdpNameSearch)) {
+                    filteredIdp = idp;
+                } else if (idp.getIdentityProviderName().equals(testIdpName)) {
+                    notFilteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Searched Idp not found");
+            Assert.assertNull(notFilteredIdp, "Searched Incorrect Idp");
+        } else {
+            Assert.fail("Unable to find filtered identity provider");
+        }
+    }
+
+    @Test(priority = 8, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation")
+    public void testGetAllPaginatedIdPsWithEmptyFilter() throws Exception {
+
+        String filter = "";
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All paginated idp list size with empty filter : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            int filteredIdpSize = 0;
+            for (IdentityProvider idp : idpList) {
+                if (idp.getIdentityProviderName().equals(testIdpName) || idp.getIdentityProviderName()
+                        .equals(testIdpNameSearch)) {
+                    filteredIdp = idp;
+                    filteredIdpSize ++;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Filtered idps not found while calling getPaginatedIdPsInfo() with " +
+                    "empty filter");
+            Assert.assertEquals(filteredIdpSize, 2, "Unable to find filtered identity provider while testing " +
+                    "getPaginatedIdPsInfo() with empty filter");
+        } else {
+            Assert.fail("Unable to find filtered identity provider while testing getPaginatedIdPsInfo().");
+        }
+    }
+
+    @Test(priority = 9, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation with invalid filter")
+    public void testGetPaginatedIdPsInfoWithInvalidFilter() {
+
+        try {
+            String filter = "names sp \"Search\" and namees ew \"Test\"";
+            int pageNumber = 1;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+            Assert.assertNull(idpList, "Idp list found while calling getPaginatedIdPsInfo() with invalid filter.");
+        } catch (Exception e) {
```

Comment:
add exception to method signature

Reply:
It is a negative test case. So we need to catch the exception